lmnet/lmnet/networks/object_detection/lm_fyolo.py

```diff
@@ -167,74 +168,38 @@ def __init__(
-
+        BaseQuantize.__init__(
+            self,
+            activation_quantizer,
+            activation_quantizer_kwargs,
+            weight_quantizer,
+            weight_quantizer_kwargs,
+            quantize_first_convolution,
+            quantize_last_convolution,
+        )
         assert callable(weight_quantizer)
         assert callable(activation_quantizer)
 
-        self.weight_quantization = weight_quantizer(**weight_quantizer_kwargs)
-        self.activation = activation_quantizer(**activation_quantizer_kwargs)
-
         if self.quantize_last_convolution:
             self.before_last_activation = self.activation
         else:
             self.before_last_activation = lambda x: tf.nn.leaky_relu(x, alpha=0.1, name="leaky_relu")
 
-    @staticmethod
-    def _quantized_variable_getter(
-            weight_quantization,
-            quantize_first_convolution,
-            quantize_last_convolution,
-            getter,
-            name,
-            *args,
-            **kwargs):
-        """Get the quantized variables.
-
-        Use if to choose or skip the target should be quantized.
-
-        Args:
-            getter: Default from tensorflow.
-            name: Default from tensorflow.
-            weight_quantization: Callable object which quantize variable.
-            args: Args.
-            kwargs: Kwargs.
-        """
-        assert callable(weight_quantization)
-        var = getter(name, *args, **kwargs)
-        with tf.variable_scope(name):
-            if "kernel" == var.op.name.split("/")[-1]:
-
-                if not quantize_first_convolution:
-                    if var.op.name.startswith("block_1/"):
-                        return var
-
-                if not quantize_last_convolution:
-                    if var.op.name.startswith("block_last/"):
-                        return var
-
-                # Apply weight quantize to variable whose last word of name is "kernel".
-                quantized_kernel = weight_quantization(var)
-                tf.summary.histogram("quantized_kernel", quantized_kernel)
-                return quantized_kernel
-
-        return var
-
-    def base(self, images, is_training):
+#    def base(self, images, is_training):
         custom_getter = partial(
             self._quantized_variable_getter,
-            self.weight_quantization,
-            self.quantize_first_convolution,
-            self.quantize_last_convolution,
+            weight_quantization=self.weight_quantization,
+            first_layer_name="block_1/",
+            last_layer_name="block_last/",
+            quantize_first_convolution=self.quantize_first_convolution,
+            quantize_last_convolution=self.quantize_last_convolution,
+            use_histogram=True
         )
-        with tf.variable_scope("", custom_getter=custom_getter):
-            return super().base(images, is_training)
+#        with tf.variable_scope("", custom_getter=custom_getter):
```

Comment:
Why do you comment out?
The network not quantized, I thiknk

Reply:
I misunderstood about that there is base class define the `custom_getter` for the `tf.variable_scope` and think there is no need to define it here.
I'm gonna fix it soon.